src/main/scala/mesosphere/marathon/core/deployment/impl/TaskReplaceActor.scala

```diff
@@ -282,22 +223,29 @@ object TaskReplaceActor extends StrictLogging {
   )
 
   /** Encapsulates the logic how to get a Restart going */
-  private[impl] case class RestartStrategy(nrToKillImmediately: Int, maxCapacity: Int)
+  private[impl] case class RestartStrategy(nrToKillImmediately: Int, nrToStartImmediately: Int)
+
+  private[impl] def computeRestartStrategy(runSpec: RunSpec, new_running: Int, old_running: Int, new_failing: Int, old_failing: Int): RestartStrategy = {
```

Comment:
Note to self: I should encapsulater the 4 ints in an object like "TransitionState" : it will be more easily readable and testable